src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/HeaderModelBinder.cs

```diff
@@ -36,35 +37,117 @@ public HeaderModelBinder()
         /// </summary>
         /// <param name="loggerFactory">The <see cref="ILoggerFactory"/>.</param>
         public HeaderModelBinder(ILoggerFactory loggerFactory)
```

Comment:
Since this was newly added for preview1, should we remove this one entirely? Presumably we want users to use the new behavior going forward.

Reply:
This constructor would be used when users are on 2.0 compat mode. We could have not added this constructor but kept it as they could see logging from the binder.